File: NoteApplicationMVPKotlin/app/src/prod/java/com.example.rafaelsavaris.noteapplicationmvp/Injection.java
package com.example.rafaelsavaris.noteapplicationmvp;

import android.content.Context;

/**
 * Created by rafael.savaris on 18/10/2017.
 */

public class Injection {

    public static NotesRepository providesNotesRepository(Context context){

        NoteDatabase database = NoteDatabase.getInstance(context);

        return NotesRepository.getInstance(NotesRemoteDataSource.getInstance(), NotesLocalDataSource.getInstance(new AppExecutors(), database.noteDao()));

    }

}

File: NoteApplicationMVPKotlin/app/src/prod/java/com.example.rafaelsavaris.noteapplicationmvpkotlin/Injection.kt
package com.example.rafaelsavaris.noteapplicationmvpkotlin

import android.content.Context
import com.example.rafaelsavaris.noteapplicationmvpkotlin.data.NotesRepository
import com.example.rafaelsavaris.noteapplicationmvpkotlin.data.source.local.NotesDatabase
import com.example.rafaelsavaris.noteapplicationmvpkotlin.data.source.local.NotesLocalDataSource
import com.example.rafaelsavaris.noteapplicationmvpkotlin.data.source.remote.NotesRemoteDataSource
import com.example.rafaelsavaris.noteapplicationmvpkotlin.utils.AppExecutors

object Injection {

    fun provideNotesRepository(context: Context): NotesRepository {

        val database = NotesDatabase.getInstance(context)

        return NotesRepository.getInstance(NotesRemoteDataSource, NotesLocalDataSource.getInstance(AppExecutors(), database.notesDao()))

    }

}